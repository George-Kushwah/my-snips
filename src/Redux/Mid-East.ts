import { all, fork } from "@redux-saga/core/effects";
import Mysaga from "./Middleware";

export default function* rootSaga() {
  yield all([fork(Mysaga)]);
}
