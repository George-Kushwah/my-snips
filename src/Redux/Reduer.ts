import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";

export interface Apidata {
  err: "";
  load: boolean;
  data: any[];
  dbtestload: boolean;
  dbtestsucc: any[];
}
const initialState: Apidata = {
  err: "",
  load: false,
  data: [],
  dbtestload: false,
  dbtestsucc: [],
};

export const myReucerdata = createSlice({
  name: "testdata",
  initialState,
  reducers: {
    error: (state, action: PayloadAction<any>) => {
      state.load = false;
      state.err = action.payload;
    },
    load: (state) => {
      state.load = true;
    },
    getData: (state, action: PayloadAction<any>) => {
      state.load = false;
      state.data = action.payload;
    },
    dbtestload: (state) => {
      state.dbtestload = true;
      state.dbtestsucc = [];
    },
    dbtestsucc: (state, action: PayloadAction<any>) => {
      state.dbtestload = false;
      state.dbtestsucc = action.payload;
    },
  },
});
export const { error, load, getData, dbtestload, dbtestsucc } =
  myReucerdata.actions;
export default myReucerdata.reducer;
