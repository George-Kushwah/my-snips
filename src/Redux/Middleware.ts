import { call, put, takeEvery } from "@redux-saga/core/effects";
import * as actions from "./Reduer";
import CustomAxios from "../layout/Axios-list";

export default function* Mysaga() {
  try {
    yield takeEvery(actions.load.type, LoadData);
    yield takeEvery(actions.dbtestload.type, MydbloadData);
  } catch {
    yield;
  }
}

function* LoadData(): any {
  try {
    let urls: any = CustomAxios("http://localhost:3000/");
    const getData: any = yield call(() =>
      urls.get("https://jsonplaceholder.typicode.com/users")
    );
    yield put(actions.getData(getData.data ?? []));
  } catch (err: any) {
    yield put(actions.error(err.toJSON().message));
  }
}

function* MydbloadData(): any {
  try {
    let urls: any = CustomAxios("http://localhost:4500/");
    const getData: any = yield call(() => urls.get("myvalue"));
    yield put(actions.dbtestsucc(getData.data ?? []));
  } catch (err: any) {
    yield put(actions.error(err.toJSON().message));
  }
}
