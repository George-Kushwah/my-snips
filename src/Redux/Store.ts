import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "@redux-saga/core";
import { myReucerdata } from "./Reduer";
import rootSaga from "./Mid-East";

const sagaMiddlware = createSagaMiddleware();
export const store = configureStore({
  reducer: {
    testdata: myReucerdata.reducer,
  },
  middleware: (defaultmiddleware) =>
    defaultmiddleware({ serializableCheck: false }).concat(sagaMiddlware),
  devTools: true,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
sagaMiddlware.run(rootSaga);
