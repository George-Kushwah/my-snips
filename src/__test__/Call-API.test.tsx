import CallAPI from "./../layout/Call-API";
import { screen, render, prettyDOM } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { store } from "../Redux/Store";
import { describe } from "node:test";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
const queryClient = new QueryClient();

describe("Call api page check", () => {
  it("Check Parent div", () => {
    render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <CallAPI />
        </QueryClientProvider>
      </Provider>
    );
    let div1 = screen.getByTestId("Mydiv");
    expect(div1).toMatchSnapshot();
  });
  it("Check Parent div", () => {
    const renderer: any = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <CallAPI />
        </QueryClientProvider>
      </Provider>
    );
    let div1 = screen.getByTestId("Mydiv");
    console.log(prettyDOM(renderer.container.firstChild));
  });
});
