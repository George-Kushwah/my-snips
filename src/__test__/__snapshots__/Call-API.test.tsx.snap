// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Parent div 1`] = `
<div
  class="container-fluid mt-2"
  data-testid="Mydiv"
>
  <h3>
    All Item Here
  </h3>
  <div
    class="my-data"
    data-testid="childdiv"
  >
    fd
  </div>
</div>
`;
